### ISSUE TITLE
error in mapply after every tf/keras function...

### ISSUE THREAD

**Issue by viebrix** [rstudio/keras#776]
Hello, and thanks for providing this package! 

Unfortunatly, when I'm using keras/tensorflow I always get R error message that mapply with the length of 0 can not mixed up with other length.
I have installed Anaconda Distribution in Windows 7 (with for all users on this PC)
Afterwards I installed R 3.6 (but directly from Windows 7 - and not in Anaconda)
I installed R-Tools 3.5

In R:
 
`install.packages("devtools")
Sys.setenv("TAR" = "internal")
devtools::install_github("rstudio/keras",force=T)
`
Downloading GitHub repo rstudio/keras@master
:
Installing package into C:/Users/xxxxxxx//Documents/R/win-library/3.6
:
Rd warning: C:/Users/xxxxxxx/AppData/Local/Temp/RtmpgdUQNd/R.INSTALL1b38cdf1210/keras/man/reexports.Rd:33: file link 'use_virtualenv' in package 'reticulate' does not exist and so has been treated as a topic
R
:

** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
*** arch - i386
*** arch - x64
** testing if installed package can be loaded from final location
*** arch - i386
*** arch - x64
** testing if installed package keeps a record of temporary installation path
* DONE (keras)

`keras::install_keras()`

Remove all packages in environment C:\Users\xxxxxxx\Documents\.conda\envs\r-tensorflow:
environment location: C:\Users\xxxxxxx\Documents\.conda\envs\r-tensorflow
:
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Creating r-tensorflow conda environment for TensorFlow installation...
Collecting package metadata: ...working... done
Solving environment: ...working... done

 environment location: C:\Users\xxxxxxxxx\Documents\.conda\envs\r-tensorflow

  added / updated specs:
    - python=3.7

The following NEW packages will be INSTALLED
:
Downloading and Extracting Packages
:
The following packages will be SUPERSEDED by a higher-priority channel:

  certifi                                         pkgs/main --> conda-forge
werkzeug-0.15.4      | 254 KB    | ########## | 100% 
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done

Installation complete.
`library(keras)`

`is_keras_available()`

[1] FALSE

Tried also:
`install_tensorflow(version = "nightly")`
`is_keras_available()`

[1] FALSE

```
library(reticulate)
use_condaenv("r-tensorflow",` required = TRUE)

is_keras_available()
```
[1] FALSE
`py_run_string("import tensorflow as tf")`
Fehler in mapply(function(from, to) paste(lines[from:to], collapse = '\n'),  : 
  Eingaben mit Laenge 0 koennen nicht mit Eingaben anderer Laenge gemischt werden
`sessionInfo()`
R version 3.6.0 (2019-04-26)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=German_Austria.1252  LC_CTYPE=German_Austria.1252    LC_MONETARY=German_Austria.1252 LC_NUMERIC=C                    LC_TIME=German_Austria.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reticulate_1.12    tensorflow_1.13.1  keras_2.2.4.1.9001

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1        rstudioapi_0.10   whisker_0.3-2     magrittr_1.5      usethis_1.5.0     devtools_2.0.2    pkgload_1.0.2     lattice_0.20-38   R6_2.4.0          rlang_0.3.4       tools_3.6.0      
[12] grid_3.6.0        pkgbuild_1.0.3    sessioninfo_1.1.1 cli_1.1.0         withr_2.1.2       tfruns_1.4        remotes_2.0.4     assertthat_0.2.1  digest_0.6.18     rprojroot_1.3-2   crayon_1.3.4     
[23] Matrix_1.2-17     processx_3.3.1    callr_3.2.0       base64enc_0.1-3   fs_1.3.1          ps_1.3.0          curl_3.3          zeallot_0.1.0     memoise_1.1.0     glue_1.3.1        compiler_3.6.0   
[34] desc_1.2.0        generics_0.0.2    backports_1.1.4   prettyunits_1.0.2 jsonlite_1.6 

`tensorflow::tf_config()`
Fehler in mapply(function(from, to) paste(lines[from:to], collapse = '\n'),  : 
  Eingaben mit Laenge 0 koennen nicht mit Eingaben anderer Laenge gemischt werden

Please, can somebody help me further.
Thanks!

Edit: Additional Info:
`reticulate::conda_list()`
         name                                                                  python
1     ANACON~1                                  C:\\ProgramData\\Anaconda3\\python.exe
2 r-tensorflow C:\\Users\\xxxxxxxxx\\Documents\\.conda\\envs\\r-tensorflow\\python.exe

in the Anaconda Prompt I also tried:
```
conda update conda
conda update --all
```

strange is that: `conda info --envs`
only deliver:
conda environments:
base                  *  C:\ProgramData\Anaconda3

**but not**: r-tensorflow C:\\Users\\xxxxxxxxx\\Documents\\.conda\\envs\\r-tensorflow\\

Maybe I don't understand the connection between conda, anaconda, rstudio/keras and tensorflow and the environment r-tensorflow wrong, but it seems for me all the installations done correctly and tensorflow still do not deliver any data to R.

Edit2:
It seems that already python has some problem with virtual environment:
`py_config()`
Fehler in mapply(function(from, to) paste(lines[from:to], collapse = '\n'),  : 
  Eingaben mit Laenge 0 koennen nicht mit Eingaben anderer Laenge gemischt werden

` reticulate::py_versions_windows()`
        type hive           install_path                    executable_path version arch
1 PythonCore  HLM C:\\PROGRA~3\\ANACON~1 C:\\PROGRA~3\\ANACON~1\\python.exe     3.7  x64
2   Anaconda  HLM C:\\PROGRA~3\\ANACON~1 C:\\PROGRA~3\\ANACON~1\\python.exe     3.7  x64

**Comment by viebrix** [comment]
`reticulate::conda_list()`
         name                                                                  python
1     ANACON~1                                  C:\\ProgramData\\Anaconda3\\python.exe
2 r-tensorflow C:\\Users\\xxxxxxxxx\\Documents\\.conda\\envs\\r-tensorflow\\python.exe


in the Anaconda Prompt I also tried:
```
conda update conda
conda update --all
```

strange is that: `conda info --envs`
only deliver:
# conda environments:
#
base                  *  C:\ProgramData\Anaconda3

**but not**: r-tensorflow C:\\Users\\xxxxxxxxx\\Documents\\.conda\\envs\\r-tensorflow\\


Maybe I don't understand the connection between conda, anaconda, rstudio/keras and tensorflow and the environment r-tensorflow wrong, but it seems for me all the installations done correctly and tensorflow still do not deliver any data to R.

**Comment by viebrix** [comment]
It seems that Anaconda is installing a .conda directory in:
C:\Users\[username]\.conda
The rstudio/keras installation is installing a .conda directory in:
C:\Users\[username]\Documents\.conda
So they can't find each other - maybe this is the reason why reticulate can't connect to python..?

**Comment by viebrix** [comment]
Keras installation with R 3.5.3 finally worked, so it seems to be really a problem with reticulate and R 3.6

**Comment by viebrix** [comment]
Installing my fork of reticulate package, seems to solve the problem:
```
devtools::install_github("viebrix/reticulate")
library(keras)
is_keras_available()
[1] TRUE
```

**Comment by skeydan** [comment]
Did you manage to solve your problem? If so you might consider adding information for people running into related installation issues.

**Comment by viebrix** [comment]
@skeydan, yes I managed with a temporary solution.

**What I did so far:**
I found out that the problem exists in using windows+R3.6 with the reticulate package. Therefore I debugged this package and found out that the system2 call to python does not return any configuration in stdout. I forked this package and changed the source code that a call to python also read stderr and not only stdout. With this change to the source code: https://github.com/rstudio/reticulate/pull/505 I did a test and the packaged worked also on R3.6 and Windows. I did an pull request for the maintainers of the reticulate package.

**What maintainers of reticulate package maybe can do:**
Maybe they can examine my source code change and identify if the error is in reticulate, or in R3.6 or in the python version I'm using (I have no experience in the use of python from R so far, therefore I can not answer the question by myself). Afterwards they can decide if they are using my source code changes or if they ask the maintainers of R to change some (possible) errors in system2 in R3.6

**What other users can do, who experience the same error :**
For a workaround they can install keras as it is described:

```
install.packages("devtools")
devtools::install_github("rstudio/keras",force=T)
library(keras)
is_keras_available()
```
If the answer is FALSE
and 
```
library(reticulate)
py_config()
```
delivers the mapply error they can try to install my fork of the reticulate package

```
devtools::install_github("viebrix/reticulate",force=T)
library(keras)
is_keras_available()
[1] TRUE
```

Afterwards everything should work like a charm, at least in my case it did.
